tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/MetricsHelper.java

```diff
@@ -212,7 +229,431 @@
+   *     dimension of size 1 that would be squeezed. If <code>multiLabel</code> or if <code>
+   *     labelWeights</code> <code>!= null</code>, then Cx must be a single dimension.
+   * @param predictions the predictions shape (N, Cx, P1?).
+   * @param thresholds thresholds in `the range <code>[0, 1]</code>, or {@link #NEG_INF} (used when
+   *     topK is set)
+   * @param topK Optional, used only if <code>multiLabel</code>, indicates that only the top k
+   *     predictions should be considered. May be null.
+   * @param classIndex Optional, limits the prediction and labels to the specified class. The
+   *     classIndex is an integer index into the first dimension of Cx.
+   * @param sampleWeight Optional <code>Tensor</code> whose rank is either 0, or the same rank as
+   *     <code>labels</code>, and must be broadcast to <code>labels</code> (i.e., all dimensions
+   *     must be either <code>1</code>, or the same as the corresponding <code>labels</code>
+   *     dimension).
+   * @param multiLabel indicates whether multidimensional prediction/labels should be treated as
+   *     multilabel responses, or flattened into a single label. When true, the values of <code>
+   *     variablesToUpdate</code> must have a second dimension equal to the number of labels and
+   *     predictions per example, and those tensors must not be RaggedTensors.
+   * @param labelWeights tensor of non-negative weights for multilabel data. The weights are applied
+   *     when calculating TRUE_POSITIVES, FALSE_POSITIVES, TRUE_NEGATIVES, and FALSE_NEGATIVES
+   *     without explicit multilabel handling (i.e. when the data is to be flattened). May be null.
+   * @param <T> the data type for the variables
+   * @throws IllegalArgumentException If <code>predictions</code> and <code>labels</code> have
+   *     mismatched shapes, or if <code>sampleWeight</code> is not <code>null</code>and its shape
+   *     doesn't match <code>predictions</code>
+   * @return an op to update the given confusion matrix variables.
+   */
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  public static <T extends TNumber> List<Op> updateConfusionMatrixVariables(
+      Ops tf,
+      Map<ConfusionMatrixEnum, Variable<T>> variablesToUpdate,
+      Map<ConfusionMatrixEnum, Assign<T>> varInitializers,
+      Operand<T> labels,
+      Operand<T> predictions,
+      Operand<TFloat32> thresholds,
+      Integer topK,
+      Integer classIndex,
+      Operand<T> sampleWeight,
+      boolean multiLabel,
+      Operand<T> labelWeights) {
+    if (multiLabel && labelWeights != null)
+      throw new IllegalArgumentException(
+          "labelWeights for multilabel data should be handled outside of updateConfusionMatrixVariables when multiLabel is true.");
+
+    if (variablesToUpdate == null || variablesToUpdate.isEmpty()) {
+      return Collections.EMPTY_LIST;
+    }
+
+    Operand<T> tLabels = labels;
```

Comment:
This is due to the desire expressed on other PRs, not to assign new values to method parameters. Therefore, I created `tLabels` so that its value can be modified later in the method.

For example:

`tLabels = tf.gather(tLabels, tf.constant(new int[] {classIndex}), tf.constant(1));`

Reply:
👍 ah yeah, I remember that previous comment now!